src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3523,22 +3518,26 @@ protected virtual void MarkInstruction (Instruction instruction, MethodDefinitio
 			}
 		}
 
-		protected virtual bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface, TypeDefinition resolvedInterfaceType)
+		protected virtual bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface)
 		{
 			if (Annotations.IsMarked (iface))
 				return false;
 
-			if (Annotations.IsMarked (resolvedInterfaceType))
+			if (!Context.IsOptimizationEnabled (CodeOptimizations.UnusedInterfaces, type))
```

Comment:
Should not this also check for skip-unresolved when someone runs with `skip-unresolved false` and `UnusedInterfaces false` it prints the correct error?